contract-variable-monitor/src/utils.js

```diff
@@ -0,0 +1,134 @@
+const BigNumber = require('bignumber.js');
+const RollingMath = require('rolling-math');
+
+function getAbi(abiName) {
+  // eslint-disable-next-line global-require,import/no-dynamic-require
+  const { abi } = require(`../abi/${abiName}`);
+  return abi;
+}
+
+// get a list of variable getter information objects for each variable name listed for a given
+// contract in the config
+function getVariableInfo(contractConfig, currentContract, variableList, contractList) {
+  const proxyName = contractConfig.proxy;
+  const { variables } = contractConfig;
+  const info = [];
+
+  let variableNames = [];
+  if (variableList === undefined) {
+    if (proxyName === undefined) {
+      return {}; // no variables for this contract are specified in the config
+    }
+  } else {
+    variableNames = Object.keys(variables);
+  }
+
+  if (proxyName) {
+    // contract is a proxy, look up public getters for the requested variables (if any) for the
+    // contract the proxy is pointing to
+    const proxyVariables = variableList[proxyName].variables;
+    if (proxyVariables) {
+      if (variableList === undefined) {
+        // eslint-disable-next-line no-param-reassign
+        variableList = { ...proxyVariables };
+      } else {
+        // eslint-disable-next-line no-param-reassign
+        variableList = { ...variableList, ...proxyVariables };
+      }
+    }
```

Comment:
I'm uncertain what this is intended to do.  If line 29 succeeds, there's no way that the conditional on line 31 will evaluate to `true`.

Also, the variable `variableList` is not used anywhere else in this function after line 36.  Is the intention to update the `configEntries` variable in the calling function (from `agent.js`)?

It looks like line 29 should remain and lines 30-38 should be removed.

Reply:
I removed the proxy logic entirely in favor of adding an extra note in `SETUP.md` for using proxy contracts